tensorflow/core/grappler/devices.cc

```diff
@@ -58,10 +59,13 @@ int GetNumAvailableGPUs(
               cuda_compute_capability >= min_cuda_compute_capability) {
             num_eligible_gpus++;
           }
-        }
-#else
-        num_eligible_gpus++;
+#else //TENSORFLOW_USE_ROCM
+          std::string gcn_arch_name = desc->rocm_amdgpu_gcn_arch_name(); 
+          if(std::find(std::begin(FP16SupportedDevices),
+             std::end(FP16SupportedDevices), gcn_arch_name)
+             != std::end(FP16SupportedDevices)) num_eligible_gpus++;
```

Comment:
`GetNumAvailableGPUs` is a generic routine, and has nothing to do with whether or not the underlying GPU has fp16 support.  This routine should not require any changes...why was this change needed?

Reply:
This was specifically used to get the number of qualified GPUs in the CUDA Implementation.

Reply:
correct....the CUDA version gets the num availbale gpus that >= the specified cuda compute capability

https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/core/grappler/devices.h#L29-L34

CUDA compute capability is a CUDA specific concept.

If you want something comparable for AMDGPUs, you would need something like

```
// Get the number of available AMDGPUs whose "gcn_arch" matches the given value
int GetNumAvailableAMDGPUs(const std::string& gcn_arch);
```

If you want something feature specific (like number of AMDGPUs that have FP16 support), you would need something like
```
// Get the number of available AMDGPUs with support for efficient FP16 compute
int GetNumAMDGPUsWithFP16Support();
```

Overloading this routine ( `GetNumAvailableGPUs`) to return the number of AMDGPUs which have efficient FP16 support is not right.
```